### PR TITLE
[Mellanox] Update mlnx-fw-upgrade.j2 to support BF3 DPU NIC FW.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
#### Why I did it
Extend `mlnx-fw-upgrade.j2` to support the upgrade of the DPU NIC FW

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Add support for the new device to the script

#### How to verify it
Compile and install an image

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by Yakiv-Huryk — APPROVED** [review]

File: platform/mellanox/mlnx-fw-upgrade.j2 [hunk 1]
```diff
@@ -339,7 +427,13 @@ function UpgradeFW() {
     if [[ "${_FW_CURRENT}" == "${_FW_AVAILABLE}" ]]; then
         ExitSuccess "firmware is up to date"
     else
+        if [[ "${DRY_RUN}" == "${YES_PARAM}" ]]; then
```
Comment (vivekrnv) [inline platform/mellanox/mlnx-fw-upgrade.j2:424]: Why do we need this? Just for user convenience?

Comment (vivekrnv) [reply platform/mellanox/mlnx-fw-upgrade.j2:424]: Got it, used it in bfnet.sh. I think "upgrade_check" or similar name would be better than dry_run

Comment (oleksandrivantsiv) [reply platform/mellanox/mlnx-fw-upgrade.j2:424]: I don't agree with this. `--dry-run` is widely used by Linux commands for purposes like this.

File: platform/mellanox/mlnx-fw-upgrade.j2 [hunk 2]
```diff
@@ -301,6 +336,62 @@ function RunFwUpdateCmd() {
...
 }
 
+function GetAvailableFwVersion() {
+    local -r _FW_FILE="$1"
+    local -r _MST_DEVICE="$2"
+    local -r _PSID="$3"
+
+    RunCmd "${LIST_CONTENT_CMD} -i ${_FW_FILE} -d ${_MST_DEVICE} -o ${LIST_CONTENT_FILE}"
+
+    ERROR_CODE="$?"
```
Comment (nazariig) [inline platform/mellanox/mlnx-fw-upgrade.j2]: @oleksandrivantsiv why do we need it? `RunCmd` exits script if `rc != 0`

**Review by nazariig — CHANGES_REQUESTED** [review]

File: platform/mellanox/mlnx-fw-upgrade.j2 (hunk 2 above)
Comment (oleksandrivantsiv) [reply platform/mellanox/mlnx-fw-upgrade.j2]: removed, thanks

**Review by nazariig — APPROVED** [review]

**Pull request closed without merging**